source/system/structs/HashMap.ooc

```diff
@@ -6,41 +6,17 @@
  * of the MIT license.  See the LICENSE file for details.
  */
 
-getStandardEquals: func <T> (k1, k2: T) -> Bool {
-	if (T == String)
-		stringEquals(k1, k2)
-	else if (T == CString)
-		cstringEquals(k1, k2)
-	else if (T size == Pointer size)
-		pointerEquals(k1, k2)
-	else if (T size == UInt size)
-		intEquals(k1, k2)
-	else if (T size == Char size)
-		charEquals(k1, k2)
-	else
-		genericEquals(k1, k2)
-}
-
-stringEquals: func <K> (k1, k2: K) -> Bool { k1 as String equals(k2 as String) }
-
-cstringEquals: func <K> (k1, k2: K) -> Bool { k1 as CString == k2 as CString }
-
-pointerEquals: func <K> (k1, k2: K) -> Bool { k1 as Pointer == k2 as Pointer }
-
-intEquals: func <K> (k1, k2: K) -> Bool { k1 as Int == k2 as Int }
-
-charEquals: func <K> (k1, k2: K) -> Bool { k1 as Char == k2 as Char }
-
-genericEquals: func <K> (k1, k2: K) -> Bool { memcmp(k1, k2, K size) == 0 }
-
 HashEntry: cover {
-	key, value: Pointer
+	key: Pointer
+	value: Pointer
```

Comment:
I wonder about that myself. We don't do it inside methods, but all our classes seem to be declared this way, so we're sticking to that.